File: tournaments/longestWord/longestWord.cpp
std::string longestWord(std::string text) {

  std::string answer = "";
  std::string current = "";

  for (int i = 0; i < text.length(); ++i) {
    if ('a' <= text[i] && text[i] <= 'z'
        || 'A' <= text[i] && text[i] <= 'Z') {
      current += text[i];
      if (current.length() > answer.length()) {
        answer = current;
      }
    } else {
      current = "";
    }
  }

  return answer;
}

File: tournaments/longestWord/longestWord.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub longestWord {
    my ($text) = @_;

    my @words = $text =~ m/([A-Za-z]+)/g;
    my $result = '';

    foreach my $word (@words) {
        if (length($result) < length($word)) {
            $result = $word;
        }
    }

    return $result;
}
